### PR TITLE
apimachinery: remove unused ignoredConversions map in converter.

### PR DESCRIPTION
**Pull request by j2gg0s**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
After https://github.com/kubernetes/kubernetes/pull/94460 , ignoredConversions AND nameFunc is not used.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by knight42 — APPROVED** [review]
/lgtm
/ok-to-test
/assign @wojtek-t

**Comment by knight42** [comment]
@j2gg0s Could you squash the commits?

File: staging/src/k8s.io/apimachinery/pkg/conversion/converter.go [hunk 1]
```diff
@@ -44,23 +44,16 @@ type Converter struct {
 	generatedConversionFuncs ConversionFuncs
 
 	// Set of conversions that should be treated as a no-op
-	ignoredConversions        map[typePair]struct{}
 	ignoredUntypedConversions map[typePair]struct{}
-
-	// nameFunc is called to retrieve the name of a type; this name is used for the
-	// purpose of deciding whether two types match or not (i.e., will we attempt to
-	// do a conversion). The default returns the go type name.
-	nameFunc func(t reflect.Type) string
```
Comment (wojtek-t) [inline staging/src/k8s.io/apimachinery/pkg/conversion/converter.go:53]: I'm surprised that it's not used. But given it's not, let's also change the NewConverter to not take any parameter if it's always unused.

Comment (j2gg0s) [reply staging/src/k8s.io/apimachinery/pkg/conversion/converter.go:53]: It will change NewConverter's signature. Can we accept this incompatible change?

**Comment by j2gg0s** [comment]
@knight42 @wojtek-t need review and point out any place need to be modified.

File: staging/src/k8s.io/apimachinery/pkg/conversion/converter.go (hunk 1 above)
Comment (jpbetz) [reply staging/src/k8s.io/apimachinery/pkg/conversion/converter.go:53]: Signature is the same (it still takes a single `NameFunc` argument, it just doesn't do anything with it now).

Comment (j2gg0s) [reply staging/src/k8s.io/apimachinery/pkg/conversion/converter.go:53]: Current implement keep `NewConverter` signature, but ignore args.

**Pull request merged**